pkg/template/registry/templateinstance/strategy.go

```diff
@@ -124,20 +130,48 @@ func (s *templateInstanceStrategy) validateImpersonation(templateInstance *templ
 	}
 
 	if templateInstance.Spec.Requester.Username != userinfo.GetName() {
-		sar := authorizationapi.AddUserToSAR(userinfo,
-			&authorizationapi.SubjectAccessReview{
-				Action: authorizationapi.Action{
-					Verb:         "impersonate",
-					Group:        userapi.GroupName,
-					Resource:     authorizationapi.UserResource,
-					ResourceName: templateInstance.Spec.Requester.Username,
-				},
-			})
-		resp, err := s.oc.SubjectAccessReviews().Create(sar)
-		if err != nil || resp == nil || !resp.Allowed {
+		if err := util.Authorize(s.kc.Authorization().SubjectAccessReviews(), userinfo, &authorization.ResourceAttributes{
+			Namespace: templateInstance.Namespace,
+			Verb:      "assign",
```

Comment:
[shrug].  we can leave it, i just worry about someone getting the name wrong or typo'ing it somewhere.